packages/web-client/app/components/card-pay/create-merchant-workflow/merchant-customization/index.ts

```diff
@@ -0,0 +1,151 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import Layer2Network from '@cardstack/web-client/services/layer2-network';
+import { inject as service } from '@ember/service';
+import { WorkflowCardComponentArgs } from '@cardstack/web-client/models/workflow/workflow-card';
+import { didCancel, restartableTask, timeout } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+import { mostReadable, random as randomColor } from '@ctrl/tinycolor';
+
+export default class CardPayDepositWorkflowTransactionAmountComponent extends Component<WorkflowCardComponentArgs> {
+  @service declare layer2Network: Layer2Network;
+  @tracked merchantBgColor: string = randomColor().toHexString();
+  @tracked merchantName: string = '';
+  @tracked merchantId: string = '';
+  @tracked lastCheckedMerchantId = '';
+  @tracked merchantNameValidationMessage = '';
+  @tracked merchantIdValidationMessage = '';
+  @tracked merchantBgColorValidationMessage = '';
+
+  get canSaveDetails() {
+    return (
+      // all fields populated
+      this.merchantBgColor &&
+      this.merchantName &&
+      this.merchantId &&
+      // unique merchant name + valid id characters
+      this.merchantIdInputState === 'valid' &&
+      // no other validation errors
+      !this.merchantNameValidationMessage &&
+      !this.merchantBgColorValidationMessage
+    );
```

Comment:
a space of much opinion, but I lean toward avoiding comments in favour of extraction, so it could look something like this:

```
return
  this.allFieldsPopulated &&
  this.merchantIdValid &&
  this.noValidationMessages
```